mulighetsrommet-api/src/test/kotlin/no/nav/mulighetsrommet/api/utbetaling/ArrangorflateServiceTest.kt

```diff
@@ -0,0 +1,173 @@
+        result.id shouldBe utbetaling.id
+        result.beregning should beInstanceOf<UtbetalingBeregningForhandsgodkjent>()
+        (result.beregning as UtbetalingBeregningForhandsgodkjent).output.belop shouldBe 10000
+    }
+
+    test("getTilsagn should return arrangorflateTilsagn by ID") {
+        val result = arrangorflateService.getTilsagn(tilsagn.id)
+
+        result.shouldNotBeNull()
+        result.id shouldBe tilsagn.id
+        result.status.status shouldBe TilsagnStatus.GODKJENT
+    }
+
+    test("getTilsagnByOrgnr should return list of tilsagn for arrangor") {
+        val result = arrangorflateService.getTilsagnByOrgnr(ArrangorflateTestUtils.underenhet.organisasjonsnummer)
+
+        result shouldHaveSize 1
+        result[0].id shouldBe tilsagn.id
+        result[0].status.status shouldBe TilsagnStatus.GODKJENT
+    }
+
+    test("getArrangorflateTilsagnTilUtbetaling should return tilsagn for given gjennomforing and period") {
+        val periode = Periode(LocalDate.of(2024, 7, 1), LocalDate.of(2024, 8, 1))
+        val result = arrangorflateService.getArrangorflateTilsagnTilUtbetaling(GjennomforingFixtures.AFT1.id, periode)
+
+        result shouldHaveSize 1
+        result[0].id shouldBe tilsagn.id
+    }
+
+    test("getRelevanteForslag should return empty list when no forslag exists") {
+        val utbetalingDto = database.db.session {
+            queries.utbetaling.get(utbetaling.id)
+        }
+        utbetalingDto.shouldNotBeNull()
+
+        val result = arrangorflateService.getRelevanteForslag(utbetalingDto)
+
+        result shouldHaveSize 0
+    }
+
+    test("toArrFlateUtbetaling should convert a forhandsgodkjent utbetaling") {
+        runBlocking {
+            database.db.session {
+                queries.deltakerForslag.upsert(
+                    DeltakerForslag(
+                        id = UUID.randomUUID(),
+                        deltakerId = deltaker.id,
+                        endring = Melding.Forslag.Endring.Deltakelsesmengde(
+                            deltakelsesprosent = 80,
+                            gyldigFra = LocalDate.of(2024, 8, 15),
+                        ),
+                        status = DeltakerForslag.Status.VENTER_PA_SVAR,
+                    ),
+                )
+            }
+
+            val utbetalingDto = database.db.session {
+                queries.utbetaling.get(utbetaling.id)
+            }
+            utbetalingDto.shouldNotBeNull()
+
+            val result = arrangorflateService.toArrFlateUtbetaling(utbetalingDto)
+
+            result.id shouldBe utbetaling.id
+            result.status shouldBe ArrFlateUtbetalingStatus.VENTER_PA_ENDRING
+            result.beregning should beInstanceOf<Beregning.Forhandsgodkjent>()
+            val beregning = result.beregning as Beregning.Forhandsgodkjent
+            beregning.antallManedsverk shouldBe 1.0
+            beregning.belop shouldBe 10000
+            beregning.deltakelser shouldHaveSize 1
```

Comment:
Disse sjekkene med `should beInstanceOf` kan evt. endres til å bruke `.shouldBeInstanceOf`, og da kan vi benytte oss av resultatet av denne sjekken uten å måtte gjøre en ekstra cast `as Beregning.Forhandsgodkjent`:

```
        result.beregning.shouldBeInstanceOf<Beregning.Forhandsgodkjent>().should {
            it.antallManedsverk shouldBe 1.0
            it.belop shouldBe 10000
            it.deltakelser shouldHaveSize 1
        }
```